version: '3.9'

services:
    php_app:
        build:
            context: docker/php8.3-fpm
        image: app/php
        container_name: php_app
        volumes:
            - ./src:/data/site.default
        networks:
            - app-network
        depends_on:
            - db

    nginx:
        build:
            context: docker/nginx
        image: app/nginx
        container_name: nginx
        ports:
            - '${FORWARD_APP_PORT:-80}:80'
        volumes:
            - ./src:/data/site.default
            - ./log/nginx:/var/log/nginx
        networks:
            - app-network
        depends_on:
            - php_app

    db:
        image: 'mysql/mysql-server:8.0'
        ports:
            - '${FORWARD_DB_PORT:-3306}:3306'
        environment:
            MYSQL_ROOT_PASSWORD: '${DB_PASSWORD}'
            MYSQL_ROOT_HOST: "%"
            MYSQL_DATABASE: '${DB_DATABASE}'
            MYSQL_USER: '${DB_USERNAME}'
            MYSQL_PASSWORD: '${DB_PASSWORD}'
            MYSQL_ALLOW_EMPTY_PASSWORD: 1
        volumes:
            - 'app-native-php-db:/var/lib/mysql'
            - ./src/database:/data
            - ./log/mysql:/var/log/mysql
        networks:
            - app-network
        healthcheck:
            test: [ "CMD", "mysqladmin", "ping", "-p${DB_PASSWORD}" ]
            retries: 3
            timeout: 5s

networks:
    app-network:
        driver: bridge
volumes:
    app-native-php-db:
        driver: local